### PR TITLE
chore(router): add docs for sanity/router package

### PR DESCRIPTION
**Pull request by binoy14**
### Description

Adds documentation for `sanity/router` package

### What to review

- Accuracy for the documentation
- If more details are needed in anything
- If examples need more details
- Also converts some `type` to `interface` 

### Notes for release

N/A

### CODE REVIEW

File: packages/sanity/src/router/route.ts [hunk 1]
```diff
@@ -8,27 +8,87 @@ import {decodeParams, encodeParams} from './utils/paramsEncoding'
...
   scope?: string
 }
 
 /**
+ * Interface for the {@link route} object.
+ *
  * @public
  */
-export const route: {
+export interface RouteObject {
```
Comment (binoy14) [inline packages/sanity/src/router/route.ts:37]: This probably needs a better name. There is already an interface with name `Route`

File: packages/sanity/src/router/types.ts [hunk 2]
```diff
@@ -36,24 +61,67 @@ export type RouteChildren =
...
 
 /**
  * @public
  */
 export interface Router extends RouterNode {
+  /**
+   * Indicates whether this router is a route.
+   * @internal
+   */
   _isRoute: boolean
```
Comment (binoy14) [inline packages/sanity/src/router/types.ts:93]: Not sure if this is actually `@internal` but I marked it assuming the use of `_isRoute` on tsdoc it gets ignored right now. Let me know if I need to change this

**Comment by mariuslundgard** [comment]
## [WIP] Content comments

### Comments on React component parameters

I feel like some of these comments are superfluous. It's the `props` to give to the component, but that's a React thing.

Also – it would be more accurate as "Props to pass to the `Link` component."

image

**Review by rexxars — APPROVED** [review]

**Review by RitaDias — APPROVED** [review]
Let's goooo :shipit:

**Pull request merged**